### PR TITLE
Add contribution lore

### PR DESCRIPTION
**Pull request by mlangbehn**
Added information about contributing to BonnyCI, and updated README.md
accordingly.

Related-issue: #4
Signed-off-by: Matt Langbehn <matthew.langbehn@gmail.com>

### CODE REVIEW

File: contributing/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- [Commits](#commits)
+  - [Commit Message](#commit-message)
+    - [Sign-off](#sign-off)
+- [Pull Requests](#pull-requests)
+
+## BonnyCI Workflow
+We use the [GitHub flow](https://guides.github.com/introduction/flow/) for code changes.
+
+## Commits
+Changes should include small commits that produce a functional tree. Doing so helps keep the repository history clear and simple to bisectable.
```
Comment (eggshell) [inline contributing/README.md]: "Simple to bisectable" should be "simple to bisect" or similar.

File: contributing/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+### Commit Message
+Please keep commit messages concise and informative. If the commit relates to an open issue, please include a line in the commit message such as `Fixes #XXX`. When in doubt, follow [this guide](http://chris.beams.io/posts/git-commit/) on writing commit messages.
+
+#### Sign-off
+To improve tracking of who did what, all commits must have a "sign-off". This sign-off is a line at the end of the commit message certifying that you have read and agreed to the [Developers' Agreement](DEVELOPER_AGREEMENT.md). Certify that you agree by adding a sign-off to your commit (**NOTE:** Use your real name, please. No pseudonyms or anonymous contributions excepted.):
```
Comment (eggshell) [inline contributing/README.md]: Don't have punctuation inside of parentheses.

File: contributing/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+This is doable via the `-s` flag for `git-commit`:
+
+```shell
+$ git add foo
+$ git commit -s -m "Adding foo file"
```
Comment (eggshell) [inline contributing/README.md]: I personally don't like the usage of `-m` since it skips writing a commit body.

**Review by eggshell — CHANGES_REQUESTED** [review]
I've got a few small nits that I made inline. Otherwise, I really like that we're improving on the existing documentation and adding clarification or links where needed.

File: contributing/README.md (hunk 1 above)
Comment (mlangbehn) [reply contributing/README.md]: Nice catch.

File: contributing/README.md (hunk 2 above)
Comment (mlangbehn) [reply contributing/README.md]: I'm inclined to agree. @eggshell, do you agree with removing the second sentence?
```markdown
(**NOTE:** use your real name, please):
```

File: contributing/README.md (hunk 3 above)
Comment (mlangbehn) [reply contributing/README.md]: I strongly agree. I kept this example from the [existing documentation](https://github.com/BonnyCI/projman).

File: contributing/README.md (hunk 2 above)
Comment (eggshell) [reply contributing/README.md]: Yep. It gets the point across with brevity.

File: contributing/README.md (hunk 3 above)
Comment (eggshell) [reply contributing/README.md]: That's fair. Since you included the link to writing a good commit message we can let this slide.

**Review by eggshell — APPROVED** [review]

**Pull request merged**